datalad/distributed/create_sibling_github.py

```diff
@@ -0,0 +1,266 @@
+        """
+        try:
+            response = r.json()
+        except Exception as e:
+            lgr.debug('Cannot get JSON payload of %s [%s]' , r, e)
+            response = {}
+        lgr.debug('%s responded with %s %s', self.fullname, r, response)
+        if r.status_code == requests.codes.created:
+            return dict(
+                status='ok',
+                preexisted=False,
+                # perform some normalization
+                reponame=response.get('name'),
+                private=response.get('private'),
+                clone_url=response.get('clone_url'),
+                ssh_url=response.get('ssh_url'),
+                html_url=response.get('html_url'),
+                # and also return in full
+                host_response=response,
+            )
+        elif r.status_code == requests.codes.unprocessable and \
+                any('already exist' in e.get('message', '')
+                    for e in response.get('errors', [])):
+            return dict(
+                status='impossible',
+                message='repository already exists',
+                preexisted=True,
+            )
+        elif r.status_code == self.response_code_unauthorized:
+            return dict(
+                status='error',
+                message=('unauthorized: %s', response.get('message')),
+            )
+        # make sure any error-like situation causes noise
+        r.raise_for_status()
+        # catch-all
+        raise RuntimeError(f'Unexpected host response: {response}')
+
+    def repo_delete_request(self, orguser, reponame):
+        r = requests.delete(
+            urljoin(
+                self.api_url,
+                self.get_repo_info_endpoint.format(
+                    user=orguser,
+                    repo=reponame)),
+            headers=self.request_headers,
+        )
+        # make sure any error-like situation causes noise
+        r.raise_for_status()
+
+
+@build_doc
+class CreateSiblingGithub(Interface):
+    """Create dataset sibling on GitHub.org (or an enterprise deployment).
+
+    GitHub is a popular commercial solution for code hosting and collaborative
+    development. GitHub cannot host dataset content (but see LFS,
+    http://handbook.datalad.org/r.html?LFS). However, in combination with other
+    data sources and siblings, publishing a dataset to GitHub can facilitate
+    distribution and exchange, while still allowing any dataset consumer to
+    obtain actual data content from alternative sources.
+
+    In order to be able to use this command, a personal access token has to be
+    generated on the platform (Account->Settings->Developer Settings->Personal
+    access tokens->Generate new token).
+    """
+
+    _examples_ = [
+        dict(text="Use a new sibling on GIN as a common data source that is "
+                  "auto-available when cloning from GitHub",
+             code_py="""\
```

Comment:
I think, as a first basic example, a single command to just create a GitHub sibling would be helpful. This would also help to give the examples section a structure that makes it clear that the long, multi-command example is indeed a single example, not four. I just tried the command, and I was quite confused because with the lack of another example, it looks like there are four example commands, two of which use "create-sibling-gin".

Reply:
Yes, I pretty much made a protocol of some testing I did. All these commands should have more.